src/rt-app_parse_config.c

```diff
@@ -683,6 +683,74 @@ static void parse_cpuset_data(struct json_object *obj, cpuset_data_t *data)
 	log_info(PIN "key: cpus %s", data->cpuset_str);
 }
 
+static sched_data_t *parse_sched_data(struct json_object *obj, int def_policy)
+{
+	sched_data_t tmp_data;
+	char *def_str_policy;
+	char *policy;
+	int prior_def = -1;
+
+	/* Get default policy */
+	def_str_policy = policy_to_string(def_policy);
+
+	/* Get Policy */
+	policy = get_string_value_from(obj, "policy", TRUE, def_str_policy);
+	if (policy ){
+		if (string_to_policy(policy, &tmp_data.policy) != 0) {
+			log_critical(PIN2 "Invalid policy %s", policy);
+			exit(EXIT_INV_CONFIG);
+		}
+	} else {
+		tmp_data.policy = -1;
+	}
+
+	/* Get priority */
+	if (tmp_data.policy == -1)
+		prior_def = -1;
+	else if (tmp_data.policy == other)
+		prior_def = DEFAULT_THREAD_NICE;
+	else
+		prior_def = DEFAULT_THREAD_PRIORITY;
+
+	tmp_data.prio = get_int_value_from(obj, "priority", TRUE, prior_def);
+
+	/* deadline params */
+	tmp_data.runtime = get_int_value_from(obj, "dl-runtime", TRUE, 0);
+	tmp_data.period = get_int_value_from(obj, "dl-period", TRUE, tmp_data.runtime);
```

Comment:
which value should be put ? if runtime is not defined we can't do much with deadline scheduler. We can add a check when deadline scheduler is slected to make sure that runtiem/deadline/period are coherent